policy-document.tf

```
@@ -0,0 +1,78 @@
+locals {
+  has_secrets = length(var.secrets) > 0
+  has_params  = length(var.parameters) > 0
+}
+
+data "aws_iam_policy_document" "secrets_access" {
+  count = local.has_secrets ? 1 : 0
+
+  statement {
+    sid    = join("", [module.base_id.id, "SecretsAccess"])
```

The join here seems unnecessary?

```suggestion
    sid    = "${module.base_id.id}SecretsAccess"
```